specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/preview/2024-09-30-preview/vaults.json

```diff
@@ -0,0 +1,2626 @@
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/LocationParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "List of all DeletedVaults in the subscription.",
+            "schema": {
+              "$ref": "#/definitions/DeletedVaultList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "deprecated": false,
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "Gets the list of DeletedVaults in subscription": {
+            "$ref": "./examples/DeletedVaults_ListBySubscriptionId.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.RecoveryServices/locations/{location}/deletedVaults/{deletedVaultName}": {
+      "get": {
+        "tags": [
+          "DeletedVaults"
+        ],
+        "description": "Get the DeletedVault details.",
+        "operationId": "DeletedVaults_Get",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/LocationParameter"
+          },
+          {
+            "$ref": "#/parameters/DeletedVaultName"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/DeletedVault"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "deprecated": false,
+        "x-ms-examples": {
+          "Get Recovery Services DeletedVault": {
+            "$ref": "./examples/DeletedVaults_Get.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.RecoveryServices/locations/{location}/deletedVaults/{deletedVaultName}/undelete": {
```

Comment:
>undelete [](http://example.com/codeflow?start=128&length=8)

nit: recover?

Reply:
ARM team is recommending "undelete" as the action name. ARM Soft Delete Spec:
[https://microsoft.sharepoint.com/:w:/t/azureresourcemanagerteam/Edcm4qzBQXtErf_7jDjOZ-gBpbPKSHipLgO7AnM6RKZ7Vg?e=TulMEc&CID=70DECD03-0693-4BB1-9EBE-72175080CB12&wdLOR=c1328EE42-BD25-457B-943D-505360E2780F](https://microsoft.sharepoint.com/:w:/t/azureresourcemanagerteam/Edcm4qzBQXtErf_7jDjOZ-gBpbPKSHipLgO7AnM6RKZ7Vg?e=TulMEc&CID=70DECD03-0693-4BB1-9EBE-72175080CB12&wdLOR=c1328EE42-BD25-457B-943D-505360E2780F) (see API Contracts section).